addons/udes_delivery_control/models/stock_picking.py

```diff
@@ -0,0 +1,204 @@
+                    first_pickings |= delivery_control_pickings
+
+            picking.u_first_picking_ids = first_pickings
+
+    @api.depends("state", "move_lines", "picking_type_id")
+    def _compute_show_mark_as_todo(self):
+        """Override to show Mark as Todo button for draft Delivery Control pickings"""
+        # TODO: Use computed boolean field for delivery control instead of type
+        delivery_control_type = self.env.ref("udes_delivery_control.picking_type_delivery_control")
+
+        super()._compute_show_mark_as_todo()
+
+        for picking in self.filtered(lambda p: p.picking_type_id == delivery_control_type):
+            picking.show_mark_as_todo = picking.state == "draft"
+
+    def button_validate(self):
+        """Override to allow Delivery Control pickings to validate without stock moves"""
+        self.ensure_one()
+
+        # TODO: Use computed boolean field for delivery control instead of type
+        delivery_control_type = self.env.ref("udes_delivery_control.picking_type_delivery_control")
+
+        if self.picking_type_id == delivery_control_type:
+            return self.action_delivery_control_done()
+        else:
+            return super(Picking, self).button_validate()
+
+    def action_confirm(self):
+        """Override to confirm Delivery Control pickings without interacting with stock moves"""
+        # TODO: Use computed boolean field for delivery control instead of type
+        delivery_control_type = self.env.ref("udes_delivery_control.picking_type_delivery_control")
+
+        delivery_control_pickings = self.filtered(
+            lambda p: p.picking_type_id == delivery_control_type
+        )
+        other_pickings = self - delivery_control_pickings
+
+        for picking in delivery_control_pickings:
+            picking.write({"state": "confirmed"})
+
+        res = True
+
+        if other_pickings:
+            res = super(Picking, other_pickings).action_confirm()
+
+        return res
+
+    def action_delivery_control_done(self):
+        """
+        Mark Delivery Control picking as done.
+        
+        Generate a Goods In picking if Delivery Control picking not already linked with one.
+
+        Otherwise check if linked Goods in Picking is in 'Waiting Another Operation' state and
+        confirm it if so.
+        """
+        # TODO: Use computed boolean field for delivery control instead of type
+        delivery_control_type = self.env.ref("udes_delivery_control.picking_type_delivery_control")
+
+        for picking in self.filtered(lambda p: p.picking_type_id == delivery_control_type):
+            picking.write({"state": "done", "date_done": fields.Datetime.now()})
+
+            if not picking.u_goods_in_picking_id:
+                self.create_goods_in_from_delivery_control()
+            elif picking.u_goods_in_picking_id.state == "waiting":
+                picking.u_goods_in_picking_id.action_confirm()
+
+        return True
+
```

Comment:
Do we need an `action_done` override somewhere?

Reply:
Maybe. I'm doing the call to `action_delivery_control_done` inside of `button_validate`, but I suppose it is safer/more logical to have an `action_done` override too. Will adjust. :)

Reply:
I've added in an override for `action_done`. It will simply call `action_delivery_control_done` for Delivery Control pickings, and work as normal for other types.

I needed to keep in the override for `button_validate` as this checks Stock Moves, but I am now calling `action_done` in the unit tests rather than `action_delivery_control_done` directly.